Camera capture doesn't work on MacOS

### Your environment.
  - Version: 0.1.14, also tried master.
  - Operating system: MacOS 11.1

### What did you do?
Ran the webrtc and facedetection examples.

### What happened?
Both failed with the same panic:

```
❯ ./webrtc < sdp

mediadevices DEBUG: 00:18:32.060105 mediadevices.go:151:
============ Found Properties ============

=============== Constraints ==============

DeviceID: any
VideoConstraints:
  Width: 640 (ideal)
  Height: 480 (ideal)
  FrameRate: any
  FrameFormat: I420 (ideal)
AudioConstraints:
  ChannelCount: any
  Latency: any
  SampleRate: any
  SampleSize: any
  IsBigEndian: any
  IsFloat: any
  IsInterleaved: any

================ Best Fit ================

Not found
panic: failed to find the best driver that fits the constraints

goroutine 1 [running]:
main.main()
	/Users/andrein/Projects/pion/mediadevices/examples/webrtc/main.go:83 +0x8b3
```

```
❯ ./facedetection
mediadevices DEBUG: 00:11:08.300393 mediadevices.go:151:
============ Found Properties ============

=============== Constraints ==============

DeviceID: any
VideoConstraints:
  Width: 640 (ideal)
  Height: 480 (ideal)
  FrameRate: any
  FrameFormat: I420 (exact)
AudioConstraints:
  ChannelCount: any
  Latency: any
  SampleRate: any
  SampleSize: any
  IsBigEndian: any
  IsFloat: any
  IsInterleaved: any

================ Best Fit ================

Not found
panic: failed to find the best driver that fits the constraints
```

@TannerGabriel suggested that the constraints might be too strict, so I removed all of them (width, height, frame format).
I've also tried setting every frame format from https://github.com/pion/mediadevices/blob/master/pkg/frame/const.go without success.

Tried debugging the examples and noticed that Session.Properties() returns null. 

I'm wondering if this might be a MacOS Security & Privacy issue. I didn't get the "allow access to the camera" prompt when running the examples.

--- a/examples/facedetection/main.go
+++ b/examples/facedetection/main.go
@@ -8,8 +8,8 @@ import (
 
 	pigo "github.com/esimov/pigo/core"
 	"github.com/pion/mediadevices"
-	_ "github.com/pion/mediadevices/pkg/driver/camera" // This is required to register camera adapter
 	"github.com/pion/mediadevices/pkg/frame"
+	_ "github.com/pion/mediadevices/pkg/driver/camera" // This is required to register camera adapter
 	"github.com/pion/mediadevices/pkg/prop"
 )
 
@@ -77,7 +77,7 @@ func main() {
 
 	mediaStream, err := mediadevices.GetUserMedia(mediadevices.MediaStreamConstraints{
 		Video: func(c *mediadevices.MediaTrackConstraints) {
-			c.FrameFormat = prop.FrameFormatExact(frame.FormatI420)
+			c.FrameFormat = prop.FrameFormatOneOf{frame.FormatI420, frame.FormatYUY2}
 			c.Width = prop.Int(640)
 			c.Height = prop.Int(480)
 		},
@@ -97,7 +97,7 @@ func main() {
 		frame, release, err := videoReader.Read()
 		must(err)
 
-		// Since we asked the frame format to be exactly YUY2 in GetUserMedia, we can guarantee that it must be YCbCr
+		// Since we asked the frame format to be exactly I420/YUY2 in GetUserMedia, we can guarantee that it must be YCbCr
 		if detectFace(frame.(*image.YCbCr)) {
 			log.Println("Detect a face")
 		}

--- a/pkg/avfoundation/AVFoundationBind/AVFoundationBind.m
+++ b/pkg/avfoundation/AVFoundationBind/AVFoundationBind.m
@@ -140,6 +140,9 @@ STATUS frameFormatToFourCC(AVBindFrameFormat format, FourCharCode *pFourCC) {
         case AVBindFrameFormatUYVY:
             *pFourCC = kCVPixelFormatType_422YpCbCr8;
             break;
+        case AVBindFrameFormatYUY2:
+            *pFourCC = kCVPixelFormatType_422YpCbCr8_yuvs;
+            break;
         // TODO: Add the rest of frame formats
         default:
             retStatus = STATUS_UNSUPPORTED_FRAME_FORMAT;
@@ -150,12 +153,15 @@ STATUS frameFormatToFourCC(AVBindFrameFormat format, FourCharCode *pFourCC) {
 STATUS frameFormatFromFourCC(FourCharCode fourCC, AVBindFrameFormat *pFormat) {
     STATUS retStatus = STATUS_OK;
     switch (fourCC) {
-         case kCVPixelFormatType_420YpCbCr8BiPlanarFullRange:
-             *pFormat = AVBindFrameFormatNV21;
-             break;
-         case kCVPixelFormatType_422YpCbCr8:
-             *pFormat = AVBindFrameFormatUYVY;
-             break;
+        case kCVPixelFormatType_420YpCbCr8BiPlanarFullRange:
+            *pFormat = AVBindFrameFormatNV21;
+            break;
+        case kCVPixelFormatType_422YpCbCr8:
+            *pFormat = AVBindFrameFormatUYVY;
+            break;
+        case kCVPixelFormatType_422YpCbCr8_yuvs:
+            *pFormat = AVBindFrameFormatYUY2;
+            break;
          // TODO: Add the rest of frame formats
          default:
              retStatus = STATUS_UNSUPPORTED_FRAME_FORMAT;

